core/src/main/java/org/elasticsearch/rest/support/RestUtils.java

```diff
@@ -238,4 +240,21 @@ public static Pattern checkCorsSettingForRegex(String corsSetting) {
 
         return null;
     }
+
+    /**
+     * Return the CORS setting as an array of origins.
+     *
+     * @param corsSetting the CORS allow origin setting as configured by the user;
+     *                    should never pass null, but we check for it anyway.
+     * @return an array of origins if set, otherwise {@code null}.
+     */
+    public static String[] corsSettingAsArray(String corsSetting) {
+        if (Strings.isNullOrEmpty(corsSetting)) {
+            return null;
```

Comment:
Yes, empty array would be easier and more consistent with how its handled in the NettyTransport.  I'll make the change.